Doc/library/mailbox.rst

```diff
@@ -65,6 +65,10 @@ Supported mailbox formats are Maildir, mbox, MH, Babyl, and MMDF.
    :exc:`KeyError` exception if the corresponding message is subsequently
    removed.
 
+   .. versionchanged:: 3.10
+      Constructor accepts a :term:`path-like object` for *path* (*dirname* in
+      Maildir).
```

Comment:
Can you add the same note at the `Maildir` documentation? Other classes are a subclass of one or the other so it should be find but I wouldn't expect users to look at Mailbox.path to know whether Maildir.dirname accepts path objects.

Reply:
I'm not sure I undetsand the rationale here?  Maildir is a subclass of Mailbox and just up-calls with 'dirname' passed directly to 'path'.  I'm thoroughly confused as to why the argument name is different in Maildir - the code is otherwise identical to all the other sub-classes and as a directory is still a path it makes no sense to me.

Reply:
I think the two classes are here mostly for historical reasons, it just looks weird to me to document the children class (Maildir) in the parent (MailBox).

Reply:
It also looks weird to me for Maildir to call the parameter *dirname* when everything else calls it *path*.

> I think the two classes

There is no "two classes here" - all of the specialisations (Maildir, mbox, MH, Babyl and MMDF) derive directly from the Mailbox abstract base class (which is where the only code change is needed).  The only reason Maildir is a special case is because the parameter, which is passed directly to Mailbox as the 'path' argument, is given a different name in its constructor for no apparent reason (it's not treated any differently in the code to the other sub-classes and a directory *is* still a path).

I think I can make the wording less clumsy removing the need to refer to the parameters by name, which will solve it.  The alternative is to specify it for all of the sub-classes which seems equally weird for a change to just the base-class.  Thanks for the feedback :)

Reply:
I've added a test for construction with a string, as I've updated all of mailbox and the tests to now use pathlib internally (which naturally led to testing with a Path).

Existing tests, and the new construction test, all passing.